presto-product-tests-utils/src/main/java/com/facebook/presto/tests/querystats/HttpQueryStatsClient.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.tests.querystats;
+
+import com.facebook.presto.execution.QueryStats;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.Response;
+import io.airlift.http.client.ResponseHandler;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Optional;
+
+import static io.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static io.airlift.http.client.Request.Builder.prepareGet;
+import static io.airlift.http.client.ResponseHandlerUtils.propagate;
+
+/**
+ * Implementation of {@link QueryStatsClient} based on Apache http client.
+ */
+public class HttpQueryStatsClient
+        implements QueryStatsClient
+{
+    private final HttpClient httpClient;
+    private final ObjectMapper objectMapper;
+    private final URI baseUri;
+
+    public HttpQueryStatsClient(HttpClient httpClient, ObjectMapper objectMapper, URI baseUri)
+    {
+        this.httpClient = httpClient;
+        this.objectMapper = objectMapper;
+        this.baseUri = baseUri;
+    }
+
+    @Override
+    public Optional<QueryStats> getQueryStats(String queryId)
+    {
+        URI uri = uriBuilderFrom(baseUri).appendPath("/v1/query").appendPath(queryId).build();
+        Request request = prepareGet().setUri(uri).build();
+        return httpClient.execute(request, new GetQueryStatsResponseHandler());
+    }
+
+    private final class GetQueryStatsResponseHandler
```

Comment:
Use `FullJsonResponseHandler` instead of a custom handler. See how `StatementClient` does it.